cluster.go

```diff
@@ -43,6 +43,74 @@ const (
 	reBootstrapInterval = 30 * time.Second
 )
 
+type rpcClientWrapper struct {
+	rpcClient *rpc.Client
+	tc        *TimeCache
+}
+
+func (rpcW *rpcClientWrapper) CallContext(
+	ctx context.Context,
+	dest peer.ID,
+	svcName, svcMethod string,
+	args, reply interface{},
+) error {
+	authErr := fmt.Sprintf("client does not have permissions to this method, service name: %s, method name: %s", svcName, svcMethod)
+	if rpcW.tc.Has(dest, svcName+svcMethod) {
+		logger.Debugf("avoiding the request since it was unauthorized recently: peer %s, service name: %s, method name: %s", dest, svcName, svcMethod)
+		return errors.New(authErr)
+	}
+
+	err := rpcW.rpcClient.CallContext(ctx, dest, svcName, svcMethod, args, reply)
+	if err != nil && err.Error() == authErr {
+		rpcW.tc.Add(dest, svcName+svcMethod)
+	}
+
+	return err
+}
+
+func (rpcW *rpcClientWrapper) MultiCall(
```

Comment:
Same as in rpc library

Reply:
Do not re-implement multicall. This is not a wrapper. This is a re-implementation.

Reply:
`MultiCall` uses wrapped `CallContext`

Reply:
Exactly, that's why it is re-implementing rpc.Multicall rather than wrapping it.

Reply:
I have dropped the struct. It's a cluster method now.